sgeop/simplify.py

```diff
@@ -198,162 +248,175 @@ def simplify_singletons(
+    _np_clust_2 = np_clusters["non_planar_cluster"] == 2
+    # ---> MARTIN ---> what does does it mean to be 'under non-planar'? ###############
+    #   Is that literally planat network edges that are underneath non-planar ones? ###
+    artifacts_under_np = np_clusters[_np_clust_2].dissolve("comp", as_index=False)
 
+    # Determine typology dispatch if artifacts are present
     if not artifacts_w_info.empty:
-        to_drop = (
-            artifacts_w_info.drop_duplicates("comp")
-            .query("solution == 'drop_interline'")
-            .drop_id
-        )
+        sol_drop = "solution == 'drop_interline'"
+        sol_iter = "solution == 'iterate'"
 
+        # Determine artifacts and road edges to drop
+        _to_drop = artifacts_w_info.drop_duplicates("comp").query(sol_drop).drop_id
+        _drop_roads = roads.drop(_to_drop.dropna().values)
+
+        # Re-run node cleaning on subset of fresh road edges
         roads_cleaned = remove_false_nodes(
-            roads.drop(to_drop.dropna().values),
+            _drop_roads,
             aggfunc={
                 "coins_group": "first",
                 "coins_end": lambda x: x.any(),
                 "_status": _status,
             },
         )
-        merged_pairs = artifacts_w_info.query("solution == 'drop_interline'").dissolve(
-            "comp", as_index=False
-        )
 
-        sorted_by_node_count = artifacts_w_info.sort_values(
-            "node_count", ascending=False
-        )
-        first = sorted_by_node_count.query("solution == 'iterate'").drop_duplicates(
-            "comp", keep="first"
-        )
-        second = sorted_by_node_count.query("solution == 'iterate'").drop_duplicates(
-            "comp", keep="last"
-        )
+        # Isolate drops to create merged pairs
+        merged_pairs = artifacts_w_info.query(sol_drop).dissolve("comp", as_index=False)
 
-        first = pd.concat(
-            [first, np_clusters[~np_clusters.non_planar]], ignore_index=True
-        )
+        # Sort artifacts by their node count low-to-high
+        sorted_node_count = artifacts_w_info.sort_values("node_count", ascending=False)
+
+        # Isolate artifacts to process as singletons – first instance
+        _1st = sorted_node_count.query(sol_iter).drop_duplicates("comp", keep="first")
+        _planar_clusters = np_clusters[~np_clusters["non_planar"]]
+        _1st = pd.concat([_1st, _planar_clusters], ignore_index=True)
 
+        # Isolate artifacts to process as singletons – last instance
+        _2nd = sorted_node_count.query(sol_iter).drop_duplicates("comp", keep="last")
+
+        # Isolate artifacts to process as clusters
         for_skeleton = artifacts_w_info.query("solution == 'skeleton'")
+
+    # Otherwise instantiate artifact containers as empty
     else:
         merged_pairs = pd.DataFrame()
-        first = pd.DataFrame()
-        second = pd.DataFrame()
+        _1st = pd.DataFrame()
+        _2nd = pd.DataFrame()
```

Comment:
why??

Reply:
More readable for me, but I have no problem reverting.